gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/spanner/SpannerOptionsStubClassComposer.java

```diff
@@ -0,0 +1,376 @@
+    VariableExpr variableExpr = VariableExpr.builder()
+        .setIsDecl(true)
+        .setScope(ScopeNode.PRIVATE)
+        .setIsFinal(true)
+        .setVariable(variable)
+        .build();
+
+    // create the constructor method
+    VariableExpr thisVariableExpr = createVarExprFromRefThisExpr(variable, classRef);
+    AssignmentExpr thisAssignExpr = createAssignmentExpr(thisVariableExpr,
+        VariableExpr.withVariable(variable));
+    // TODO : explore what's a clean strategy to pass datatypes instead of getBoxedTypeWithFallBack
+    MethodDefinition constructor =
+        MethodDefinition.constructorBuilder()
+            .setArguments(
+                Arrays.asList(VariableExpr.builder()
+                    .setIsDecl(true)
+                    .setVariable(Variable.builder().setName(nameInLoweCamelCase)
+                        .setType(getBoxedTypeWithFallBack(field.getKey().type())).build())
+                    .build()))
+            .setScope(ScopeNode.PUBLIC)
+            .setBody(
+                Arrays.asList(ExprStatement.withExpr(thisAssignExpr)))
+            .setReturnType(TypeNode.withReference(classRef))
+            .build();
+
+    // create append method
+    // options.maxBatchingDelayMs = maxBatchingDelayMs;
+    VariableExpr optionNameFromOptionsClass = fieldFromParentObjectExpr(OPTIONS_VAR, variable);
+    AssignmentExpr optionsAssignExpr = createAssignmentExpr(optionNameFromOptionsClass,
+        VariableExpr.withVariable(variable));
+
+    MethodDefinition appendToOptions =
+        MethodDefinition.builder()
+            .setAnnotations(Arrays.asList(AnnotationNode.OVERRIDE))
+            .setArguments(VariableExpr.builder()
+                .setIsDecl(true)
+                .setVariable(OPTIONS_VAR)
+                .build())
+            .setReturnType(TypeNode.VOID)
+            .setName("appendToOptions")
+            .setScope(ScopeNode.PUBLIC)
+            .setBody(
+                Arrays.asList(ExprStatement.withExpr(optionsAssignExpr)))
+            .build();
+
+    ClassDefinition classDefinition =
+        ClassDefinition.builder()
+            .setImplementsTypes(Arrays.asList(getOptionsGroupTypeNode(field.getValue())))
+            .setExtendsType(TypeNode.withReference(INTERNAL_OPTION_REF))
+            .setScope(ScopeNode.PROTECTED)
+            .setIsStatic(true)
+            .setIsFinal(true)
+            .setName(nameInUpperCamelCase)
+            .setIsNested(true)
+            .setStatements(Arrays.asList(ExprStatement.withExpr(variableExpr)))
+            .setMethods(Arrays.asList(constructor, appendToOptions))
+            .build();
+    return classDefinition;
+  }
+
+  private static TypeNode getOptionsGroupTypeNode(final String value) {
+    // TODO add other mapping groups to the below switch case
+    switch (value) {
+      case "TransactionOption" :
+        return TypeNode.withReference(TRANSACTION_OPTION);
```

Comment:
@blakeli0 I had a doubt on this. This method returns a `TypeNode` that is used to specify the interfaces to implement. For ex - `implements Options.TransactionOption` . Now `TransactionOption` is supposed to be an inner interface with the `Options` class. How can we do such chaining of classes while creating a type node?